File: Data Structures & Algorithms/reverse-string/submission-1.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        i = 0
        j = len(s) - 1
        while(i<=j):
            new = s[i]
            s[i] = s[j]
            s[j] = new
            i+=1
            j-=1
        

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.java
class Solution {
    public int maxProfit(int[] prices) {
        //Bruteforce solution 
        // int maxProfit = 0;
        // for(int i = 0; i<prices.length;i++){
        //     for(int j = i;j<prices.length;j++){
        //         if(prices[j]-prices[i]>0){
        //             maxProfit = Math.max(maxProfit,prices[j]-prices[i]);
        //         }
        //     }
        // }
        // return maxProfit;

        int l = 0;
        int maxProfit = 0;
        for(int r = 1;r<prices.length;r++){
            if(prices[l]>prices[r]){
                l = r;
            }
            if(prices[r] - prices[l] > 0){
                maxProfit = Math.max(maxProfit,prices[r] - prices[l]);
            }
        }
        return maxProfit;
    }
}
